client/src/context/ProfileContext.js

```diff
@@ -0,0 +1,163 @@
+	switch (action.type) {
+	case types.Set_Is_Loading:
+		return { ...state, isLoading: true };
+	case types.Set_Error:
+		return { ...state, isLoading: false, error:action.payload };
+	case types.Set_All_Profiles:
+		return { ...state, allProfiles: action.payload, isLoading: false };
+	case types.Set_Profile:
+		return { ...state, profile: action.payload, isLoading: false };
+	case types.Clear_Profile:
+		return { ...state, profile: null, isLoading: false };
+	// case types.Delete_Profile:
+	// 	return { ...state, allProfiles: state.allProfiles.filter((profile) => profile.id !== action.payload), loading: false };
+	default:
+		return state;
+	}
+};
+
+//Context has Provider and Consumer in itself.
+//Here we set Context Provider and stored our states and actions in it to provide them to Consumer
+
+const ProfileState = (props) =>{
+
+	const initialState ={
+		allProfiles:[],
+		profile:null,
+		isLoading:false,
+		error:null,
+	};
+
+	const [state, dispatch] = useReducer(profileReducer, initialState);
+	const baseUrl ='https://jsonplaceholder.typicode.com/users';
+	const getAllProfiles = ()=>{
+		dispatch({ type: types.Set_Is_Loading }),
+		axios.get(`${baseUrl}`)
+			.then((response)=>{
+				console.log(response);
+				dispatch({ type: types.Set_All_Profiles, payload:response.data });
+			})
+			.catch((error)=>{
+				dispatch({ type:types.Set_Error, payload:error });
+			});
+	};
+
+
+	const getProfile =  (id) => {
+		dispatch({ type: types.Set_Is_Loading });
+		axios.get(`${baseUrl}/${id}`)
+			.then((response)=>{
+				dispatch({ type: types.Set_Profile, payload:response.data });
+			})
+			.catch((error)=>{
+				dispatch({ type:types.Set_Error, payload:error });
+			});
+	};
+
+
+	const addProfile = (profile) => {
+		dispatch({ type: types.Set_Is_Loading });
+		const config = {
+			headers: {
+				'Content-Type': 'application/json',
+			},
+		};
+
+		axios.post(`${baseUrl}`, profile, config)
+			.then((response)=>{
+				dispatch({ type: types.Set_Profile, payload:response.data });
+			})
+			.catch((error)=>{
+				dispatch({ type:types.Set_Error, payload:error });
+			});
+	};
+
+
+	// const deleteProfile = (id) => {
+	// 	dispatch({ type: types.Set_Is_Loading });
+	// 	axios.delete(`https://jsonplaceholder.typicode.com/users/${id}`)
+	// 		.then((response)=>{
+	// 			dispatch({ type: types.Set_Profile, payload:response.data });
+	// 		})
+	// 		.catch((error)=>{
+	// 			dispatch({ type:types.Set_Error, payload:error });
+	// 		});
+	// };
+
+	const editProfile = (profile) => {
+		dispatch({ type: types.Set_Is_Loading });
+		const config = {
+			headers: {
+				'Content-Type': 'application/json',
+			},
+		};
+
+		axios.put(`${baseUrl}/${profile.id}`, profile, config)
+			.then((response)=>{
+				dispatch({ type: types.Set_Profile, payload:response.data });
+			})
+			.catch((error)=>{
+				dispatch({ type:types.Set_Error, payload:error });
+			});
+	};
+
+	const clearProfile = ()=>{
+		dispatch({ type: types.Clear_Profile });
```

Comment:
As above, you could save a few lines and reuse your `Set_Profile` action by passing `null` as the payload in your action:
```suggestion
		dispatch({ type: types.Set_Profile, payload: null });
```